### PR TITLE
fix: [CDS-74678]: advanced step default values from step class

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- Populate advanced step default values in right drawer

#### Screenshots

https://github.com/harness/harness-core-ui/assets/106532291/72769bef-77ef-4e6b-ae32-590c7ff7f552

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by rameshx — APPROVED** [review]

**Review by PratyushGar — DISMISSED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/AdvancedSteps/__tests__/AdvancedSteps.test.tsx [hunk 1]
```diff
@@ -38,12 +40,68 @@ describe('<AdvancedSteps /> tests', () => {
...
+                  ]
+                }))
+              }))
+            } as any
+          }
+        />
+      </TestWrapper>
+    )
+    const failureStrategyPanel = getByTestId('failureStrategy-summary')
+    failureStrategyPanel.click()
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineSteps/AdvancedSteps/__tests__/AdvancedSteps.test.tsx:97]: use `userEvent.click()`

File: src/modules/70-pipeline/components/PipelineSteps/AdvancedSteps/__tests__/AdvancedSteps.test.tsx [hunk 2]
```diff
@@ -38,12 +40,68 @@ describe('<AdvancedSteps /> tests', () => {
...
+
+    await waitFor(() => {
+      expect(
+        (queryByNameAttribute('failureStrategies[0].onFailure.action.spec.timeout', container) as HTMLInputElement)
+          .value
+      ).toBe('2h')
+      expect(queryByText('pipeline.failureStrategies.errorTypeLabels.Verification')).toBeInTheDocument()
```
Comment (rameshx) [inline src/modules/70-pipeline/components/PipelineSteps/AdvancedSteps/__tests__/AdvancedSteps.test.tsx:104]: use `getBy*`. `queryBy*` is useful for asserting elements that are not present

File: src/modules/70-pipeline/components/PipelineSteps/AdvancedSteps/__tests__/AdvancedSteps.test.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineSteps/AdvancedSteps/__tests__/AdvancedSteps.test.tsx]: done

File: src/modules/70-pipeline/components/PipelineSteps/AdvancedSteps/__tests__/AdvancedSteps.test.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineSteps/AdvancedSteps/__tests__/AdvancedSteps.test.tsx]: done

**Review by rameshx — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Review by harshilbits — APPROVED** [review]

**Pull request merged**